platform/posix/transport/src/openssl_posix.c

```diff
@@ -0,0 +1,674 @@
+                        ( int32_t ) pOpensslCredentials->sniHostNameLen,
+                        pOpensslCredentials->sniHostName ) );
+        }
+
+        /* Free the malloc'd SNI host name. */
+        if( sniHostName != NULL )
+        {
+            free( sniHostName );
+        }
+    }
+}
+
+OpensslStatus_t Openssl_Connect( NetworkContext_t pNetworkContext,
+                                 const ServerInfo_t * pServerInfo,
+                                 const OpensslCredentials_t * pOpensslCredentials,
+                                 uint32_t sendTimeoutMs,
+                                 uint32_t recvTimeoutMs )
+{
+    SocketStatus_t socketStatus = SOCKETS_SUCCESS;
+    OpensslStatus_t returnStatus = OPENSSL_SUCCESS;
+    long verifyPeerCertStatus = X509_V_OK;
+    int sslStatus = 0;
+    SSL_CTX * pSslSetup = NULL;
+
+    /* Validate parameters. */
+    if( pNetworkContext == NULL )
+    {
+        LogError( ( "Parameter check failed: pNetworkContext is NULL." ) );
+        returnStatus = OPENSSL_INVALID_PARAMETER;
+    }
+    else if( pOpensslCredentials == NULL )
+    {
+        LogError( ( "Parameter check failed: pOpensslCredentials is NULL." ) );
+        returnStatus = OPENSSL_INVALID_PARAMETER;
+    }
+    else
+    {
+        /* Empty else. */
+    }
+
+    /* Establish the TCP connection. */
+    if( returnStatus == OPENSSL_SUCCESS )
+    {
+        socketStatus = Sockets_Connect( &pNetworkContext->socketDescriptor,
+                                        pServerInfo,
+                                        sendTimeoutMs,
+                                        recvTimeoutMs );
+
+        if( socketStatus == SOCKETS_INVALID_PARAMETER )
+        {
+            returnStatus = OPENSSL_INVALID_PARAMETER;
+        }
+        else if( socketStatus == SOCKETS_DNS_FAILURE )
+        {
+            returnStatus = OPENSSL_DNS_FAILURE;
+        }
+        else if( socketStatus == SOCKETS_CONNECT_FAILURE )
+        {
+            returnStatus = OPENSSL_CONNECT_FAILURE;
+        }
+        else
+        {
+            /* Empty else. */
+        }
+    }
+
+    /* Setup for creating a TLS client. */
+    if( returnStatus == OPENSSL_SUCCESS )
+    {
+        pSslSetup = SSL_CTX_new( TLS_client_method() );
+
+        if( pSslSetup == NULL )
+        {
+            LogError( ( "Creation of a new SSL_CTX object failed." ) );
+            returnStatus = OPENSSL_API_ERROR;
+        }
+        else
+        {
+            /* Set auto retry mode for the blocking calls to SSL_read and SSL_write.
+             * The mask returned by SSL_CTX_set_mode does not need to be checked. */
+            ( void ) SSL_CTX_set_mode( pSslSetup, SSL_MODE_AUTO_RETRY );
+        }
+    }
+
+    /* Set credentials for the TLS handshake. */
+    if( returnStatus == OPENSSL_SUCCESS )
+    {
+        sslStatus = setCredentials( pSslSetup,
+                                    pOpensslCredentials );
+
+        /* Setup authentication. */
```

Comment:
Misplaced comment.

Reply:
hmm not sure why all these comments are in the wrong place